scipy/stats/_hypotests.py

```diff
@@ -1946,7 +1946,7 @@ def _tukey_hsd_iv(args, equal_var):
     return args
 
 
-@xp_capabilities(np_only=True)
+@xp_capabilities(out_of_scope=True)
```

Comment:
`statistic` and `pvalue` are computed pairwise and stored in 2D matrices given 1D slices of different samples, so adding the `_axis_nan_policy` decorator is out given the existing interface. Also, almost all the time is spent computing the p-value using the studentized range distribution, but that's currently integrated using a `LowLevelCallable` for speed. Making meaningful use of array API here would be a pretty big project.